### DISCUSSION TITLE
Multiple cursor case preserve

### DISCUSSION THREAD

**Discussion by LewisAlderman** [zed-industries/zed#25480] [Feature Requests]
### Summary

Preserve the casing of the multiple individual selections whilst being edited.

References:
https://marketplace.visualstudio.com/items?itemName=Cardinal90.multi-cursor-case-preserve

![Example gif](https://raw.githubusercontent.com/Cardinal90/multi-cursor-case-preserve/master/images/Example.gif)

**Comment by JosephTLyons** [comment]
This would be such a delightful feature in Zed!

**Comment by recite-me-richard** [comment]
I was constantly using that extension in VSCode before moving to Zed, so having this feature in Zed would be amazing.

**Comment by sebastienbarre** [comment]
Big fan of this extension in vscode too.

Semi-related to that one, I created #41070, maybe you guys know how to solve this issue? Thank you

**Comment by felixgeissler** [comment]
Would love if this lands in Zed ❤️

**Comment by Nemesis19** [comment]
Me too! such a useful feature this would be!

**Comment by napalmpapalam** [comment]
any updates? 2 years issue

**Comment by emizzle** [comment]
This would be a very welcome addition 🙏

**Comment by RevolvingMadness** [comment]
I need this 🙏🏻
